sys/kern/sleepq.c

```diff
@@ -216,18 +216,33 @@ static void sq_leave(thread_t *td, sleepq_chain_t *sc, sleepq_t *sq) {
 static int _sleepq_wait(void *wchan, const void *waitpt, sleep_t sleep) {
   thread_t *td = thread_self();
   int status = 0;
+#define EARLY_INTR(td, sleep)                                                  \
```

Comment:
Generally speaking, we avoid macro definitions in procedure scope. 
You could've choosen a better name for it as well.
Maybe `TD_INTERRUPTED` ?

Reply:
Ok, I'll make it a static function, maybe something like `_sleepq_interrupted_early`?

Reply:
Seems ok.